services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ProtectedPathSpec.groovy

```diff
@@ -1105,6 +1107,60 @@ class ProtectedPathSpec extends HealthCheckSpecification {
         flowHelper.deleteFlow(flow.id)
     }
 
+    @Ignore("https://github.com/telstra/open-kilda/issues/2762")
+    def "System reuses current protected path when can't find new non overlapping protected path while intentional\
+ rerouting"() {
+        given: "Two active neighboring switches with three diverse paths"
+        def switchPair = topologyHelper.getAllNeighboringSwitchPairs().find {
+            it.paths.unique(false) { a, b -> a.intersect(b) == [] ? 1 : 0 }.size() == 3
+        } ?: assumeTrue("No suiting switches found", false)
+
+        and: "A flow with protected path"
+        def flow = flowHelper.randomFlow(switchPair)
+        flow.allocateProtectedPath = true
+        flowHelper.addFlow(flow)
+
+        def flowPathInfo = northbound.getFlowPath(flow.id)
+        assert flowPathInfo.protectedPath
+
+        def currentPath = pathHelper.convert(flowPathInfo)
+        def currentProtectedPath = pathHelper.convert(flowPathInfo.protectedPath)
+        assert currentPath != currentProtectedPath
+
+        when: "Make the current and protected path less preferable than alternatives"
+        def alternativePaths = switchPair.paths.findAll { it != currentPath && it != currentProtectedPath }
+        alternativePaths.each { pathHelper.makePathMorePreferable(it, currentPath) }
+        alternativePaths.each { pathHelper.makePathMorePreferable(it, currentProtectedPath) }
+
+        and: "Init intentional reroute"
+        def rerouteResponse = northbound.rerouteFlow(flow.id)
+
+        then: "Flow is rerouted"
+        rerouteResponse.rerouted
+
+        and: "Main path is not changed to protected path"
```

Comment:
```
and: "Init intentional reroute"
then: "Flow is rerouted"
and: "Main path is not changed to protected path"
and: "Protected path is not changed"
```
It's not quite clear from the above steps what have happened to the main path after all. Can we please name steps closer to _what has/should happen_ instead of _what has NOT happen_. For sure it's no a strong rule and we are flexible, but in this current case I don't really understand in which state my main path is expected to be

Reply:
fixed
```
and: "Init intentional reroute"
then: "Flow should be rerouted"
and: "Flow main path should be rerouted to a new path and ignore protected path"
and: "Flow protected path shouldn't be rerouted due to lack of non overlapping path"
```